alf/networks/networks.py

```diff
@@ -308,3 +315,186 @@ def forward(self, input, state):
                 lambda x: x.float() if x.dtype.is_floating_point else x, input)
         with torch.cuda.amp.autocast(self._enabled):
             return self._net(input, state)
+
+
+@alf.configurable
+@alf.repr_wrapper
+class NoisyFC(Network):
+    """The Noisy Linear Layer discribed in
+
+    Fortunato et. al. `Noisy Networks for Exploration <https://arxiv.org/abs/1706.10295>`_
+
+    Some details:
+
+    1. The noise for each sample in a batch is different.
+    2. The noise is maintained as state. It has a probability of `new_noise_prob`
+       to change to new noise.
+    3. Since the initial state is always 0, a new noise will always be generated
+       for zero state.
+    4. If it is running in eval mode (i.e., common.is_eval() is True), noise will
+       be disabled (i.e. same as alf.layers.FC).
+    5. The noise is factorized Gaussian noise as described in the paper.
+
+
+    Args:
+        Args:
+            input_size: input size.
+            output_size: output size.
+            activation: activation function.
+            std_init: the scaling factor for the initial value of weight_sigma
+                and bias_sigma.
+            new_noise_prob: the probability of resample the noise.
+            use_bn: whether use batch normalization.
+            use_ln: whether use layer normalization
+            bn_ctor: will be called as ``bn_ctor(num_features)`` to
+                create the BN layer.
+            kernel_initializer: initializer for the FC layer kernel.
+                If none is provided a ``variance_scaling_initializer`` with gain as
+                ``kernel_init_gain`` will be used.
+            kernel_init_gain: a scaling factor (gain) applied to
+                the std of kernel init distribution. It will be ignored if
+                ``kernel_initializer`` is not None.
+            bias_init_value: a constant for the initial bias value.
+                This is ignored if ``bias_initializer`` is provided.
+            bias_initializer:  initializer for the bias parameter.
+            weight_opt_args: If provided, it will be used as optimizer arguments
+                for weight. And it will be combined with zero_mean=False and
+                fixed_norm=False as optimizer arguments for weight_sigma.
+            bias_opt_args: If provided, it will be used as optimizer arguments
+                for bias. And it will be combined with zero_mean=False as
+                optimizer arguments for bias_sigma.
+    """
+
+    def __init__(self,
+                 input_size: int,
```

Comment:
Should the standard api for a network be `input_tensor_spec` instead of `input_size`?

Reply:
It can be flexible for constructor as many other networks in the same file show.